bluemira/base/constants.py

```diff
@@ -42,68 +64,218 @@
+# Tritium decay constant
+T_LAMBDA = np.log(2) / T_HALFLIFE  # [1/yr]
 
-# Helium-3 molar mass
-HE3_MOLAR_MASS = 3.0160293  # [u] or [g/mol]
+# Tritium molar mass,  [u] or [g/mol]
+T_MOLAR_MASS = elements.isotope("T").mass
 
-# neutron molar mass
-NEUTRON_MOLAR_MASS = 1.008665  # [u] or [g/mol]
+# Deuterium molar mass, [u] or [g/mol]
+D_MOLAR_MASS = elements.isotope("D").mass
 
-# proton molar mass
-PROTON_MOLAR_MASS = 1.007276466879  # [u] or [g/mol]
+# Helium molar mass, [u] or [g/mol]
+HE_MOLAR_MASS = elements.isotope("He").mass
 
-# electron molar mass
-ELECTRON_MOLAR_MASS = 5.48579909070e-4  # [u] or [g/mol]
+# Helium-3 molar mass, [u] or [g/mol]
+HE3_MOLAR_MASS = elements.isotope("3-He").mass
 
 # Absolute zero in Kelvin
 ABS_ZERO_K = 0  # [K]
 
 # Absolute zero in Celsius
-ABS_ZERO_C = -273.15  # [°C]
+ABS_ZERO_C = ureg.Quantity(0, ureg.kelvin).to("celsius").magnitude  # [°C]
 
 # =============================================================================
 # Conversions
 # =============================================================================
 
 # Electron-volts to Joules
-EV_TO_J = 1.602176565e-19
+EV_TO_J = ureg.Quantity(1, ureg.eV).to("joule").magnitude
 
 # Joules to Electron-volts
-J_TO_EV = 1 / EV_TO_J
+J_TO_EV = ureg.Quantity(1, ureg.joule).to("eV").magnitude
 
 # Atomic mass units to kilograms
-AMU_TO_KG = 1.660539040e-27
+AMU_TO_KG = ureg.Quantity(1, ureg.amu).to("kg").magnitude
 
 # Years to seconds
-YR_TO_S = 60 * 60 * 24 * 365
+YR_TO_S = ureg.Quantity(1, ureg.year).to("s").magnitude
 
 # Seconds to years
-S_TO_YR = 1 / YR_TO_S
+S_TO_YR = ureg.Quantity(1, ureg.second).to("year").magnitude
+
+
+def to_celsius(kelvin: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    kelvin: Union[float, np.array, List[float]]
+        The temperature to convert [K]
+
+    Returns
+    -------
+    temp_in_celsius: Union[float, np.array]
+        The temperature [°C]
+    """
+    if np.any(np.less(kelvin, ABS_ZERO_K)):
+        raise ValueError("Negative temperature in K specified.")
+    return ureg.Quantity(kelvin, ureg.kelvin).to("celsius").magnitude
+
+
+def to_kelvin(celsius: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    temp_in_celsius: Union[float, np.array, List[float]]
+        The temperature to convert [°C]
+
+    Returns
+    -------
+    temp_in_kelvin: Union[float, np.array]
+        The temperature [K]
+    """
+    if np.any(np.less(celsius, ABS_ZERO_C)):
+        raise ValueError("Negative temperature in K specified.")
+    return ureg.Quantity(celsius, ureg.celsius).to("kelvin").magnitude
+
+
+def from_keV(
+    value: Union[float, np.array, List[float]],
+    to: str = "celsius",
+    *,
+    _from: str = "keV"
+):
+    """
+    Convert a temperature in Celsius to Kelvin.
```

Comment:
```suggestion
    Convert a temperature in keV to Celsius.
```

Out of interest, why do we default to `Celsius` here rather than `Kelvin`?

Reply:
The reason is really just because, I'm wondering whether we change it to kelvin generally, it seems more of a standard thing to use. @CoronelBuendia any thoughts?

Reply:
The reason is really just because, I'm wondering whether we change it to kelvin generally, it seems more of a standard thing to use. @CoronelBuendia any thoughts?

Reply:
Agreed as K across the board